### PR TITLE
Navigator - Fix wrong scene transformation after pop

### LINKED ISSUE #9485: [Navigator] Wrong scene transformation after pop
Hi
I have react-native fork with some little changes in navigator example (https://github.com/sooth-sayer/react-native/tree/uiexplorer-navigator-test)

If I push scenes with different animations and then will return back to first scene (popToTop), then the first scene will be transformed incorrectly.

navigator

### PR DESCRIPTION
**Pull request by sooth-sayer**
Fixes #9485

### CODE REVIEW

**Comment by facebook-github-bot-4** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

**Comment by mkonicek** [comment]
Thanks for the great demo in https://github.com/facebook/react-native/issues/9485!

@ericvicenti Does this look good to you?

File: Libraries/CustomComponents/Navigator/Navigator.js [hunk 1]
```diff
@@ -1087,6 +1101,7 @@ var Navigator = React.createClass({
     var presentedRoute = this.state.routeStack[this.state.presentedIndex];
     var popSceneConfig = this.props.configureScene(presentedRoute); // using the scene config of the currently presented view
     this._enableScene(popIndex);
+    this._clearTransformations(popIndex);
```
Comment (mkonicek) [inline Libraries/CustomComponents/Navigator/Navigator.js]: Can you please add a comment why this is needed so people reading the code in the future understand why it's needed to reset the transform here?

**Comment by facebook-github-bot-3** [comment]
@sooth-sayer updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9516/files/6526d665257fed75338f83d112eb6ff4ffe9b39d..10c5093b74d335b3e81b302fa7d0623041c12d42)

**Comment by sooth-sayer** [comment]
@mkonicek Is this explanation (https://github.com/facebook/react-native/pull/9516/commits/10c5093b74d335b3e81b302fa7d0623041c12d42) clear enough?)

**Comment by lacker** [comment]
@sooth-sayer can you fix the problem that eslint-bot pointed out?

**Comment by sooth-sayer** [comment]
@lacker Ok, I will fix it

**Comment by sooth-sayer** [comment]
@lacker fixed

**Comment by ericvicenti** [comment]
Ok this makes sense to me.

@facebook-github-bot shipit

**Comment by ericvicenti** [comment]
@facebook-github-bot shipit

**Comment by lacker** [comment]
I think this needs a CLA to ship it. @sooth-sayer would you mind signing the CLA?

**Comment by sooth-sayer** [comment]
@lacker Ok, I signed it

**Comment by lacker** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**